### PR TITLE
Configurable restricted VPC for tooling account

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Feature

### Detail
- New cdk.json configuration parameter `tooling_vpc_restricted_nacl`
- If set to `true` we create a custom NACL for the tooling data.all created VPC with the following inbound rules. Outbound allows all traffic.

image

- We have enabled DNS private names for CodeArtifact endpoints, to correctly resolve them inside the VPC and keep traffic within the VPC.
- Modified some CodeBuild steps to install pip and npm packages from CodeArtifact instead of from the internet.
- In #323, CodeBuild Linux images were updated, eliminating the need to install yum packages and node.

### Relates
- #307 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dlpzx** [comment]
Hi @kukushking, I tested the above configuration in a fresh deployment and it completed successfully :)

File: deploy/stacks/vpc.py [hunk 1]
```diff
@@ -125,6 +126,55 @@ def create_new_vpc(self, cidr, envname, resource_prefix):
...
+                "entryInboundCustomTCP",
+                cidr=ec2.AclCidr.any_ipv4(),
+                traffic=ec2.AclTraffic.tcp_port_range(start_port=1024, end_port=65535),
+                rule_number=102,
+                direction=ec2.TrafficDirection.INGRESS,
+                rule_action=ec2.Action.ALLOW
+            )
+            nacl.add_entry(
+                "entryInboundAllInVPC",
+                cidr=ec2.AclCidr.ipv4("10.0.0.0/16"),
```
Comment (kukushking) [inline deploy/stacks/vpc.py:170]: How about referring to`self.vpc.cidr` here just in case the user changes the cidr from the default?

Comment (dlpzx) [reply deploy/stacks/vpc.py]: Good idea! I just added it

Comment (dlpzx) [reply deploy/stacks/vpc.py]: Tested 👍

**Review by kukushking — APPROVED** [review]
Looks great!

**Pull request merged**